src/userAgent.js

```diff
@@ -54,16 +56,31 @@ var UserAgent = Class(function () {
 
   if (isAndroid) {
     osType = 'Android';
-    osVersionString = ua.match(/Android[/\s][\d.]+/)[0];
-    osVersion = osVersionString.match(/[\d.]+/)[0];
+    match = ua.match(/Android[/\s][\d.]+/);
+    if (match) {
+      osVersionString = match[0];
+      osVersion = osVersionString.match(/[\d.]+/)[0];
```

Comment:
This could potentially hit a similar TypeError if the match fails, we should protect against that with something like:
```
var osVersionMatch = osVersionString.match(/[\d.]+/);
osVersion = (osVersionMatch && osVersionMatch[0]) || osVersion;
```
We could also just use an `if` if the above style is too unclear